### PR TITLE
feat: make components use window number

### PR DESCRIPTION
**Pull request by famiu**
This needs to be merged in order to make further progress on #55.

I changed quite a few things, so I expect a lot of breakages. I'd really love if people take the time to test it out before I merge this.

### CODE REVIEW

File: lua/feline/generator.lua [hunk 1]
```diff
@@ -230,25 +229,25 @@ local function parse_statusline_section_old(section, type)
...
+function M.generate_statusline(winnr)
     if not M.components then
-        return ""
+        return ''
     end
 
     local statusline_type
 
-    if is_active and not is_forced_inactive() then
+    if winnr == vim.api.nvim_get_current_win() and not is_forced_inactive() then
```
Comment (ram02z) [inline lua/feline/generator.lua:250]: This will always evaluate to false since winnr is nil

File: lua/feline/providers/lsp.lua [hunk 2]
```diff
@@ -1,59 +1,67 @@
 local M = {}
 
-function M.is_lsp_attached()
-    return next(vim.lsp.buf_get_clients()) ~= nil
+local lsp = vim.lsp
```
Comment (ram02z) [inline lua/feline/providers/lsp.lua:3]: https://github.com/famiu/feline.nvim/pull/33 was merged to stop loading it eagerly.

Comment (famiu) [reply lua/feline/providers/lsp.lua]: Oops, forgot about that. Fixed now

File: lua/feline/generator.lua (hunk 1 above)
Comment (famiu) [reply lua/feline/generator.lua:250]: Added default value for winnr to work around that for now. Once this is merged I'll make proper use and actually pass the window number to the statusline generator

**Comment by famiu** [comment]
If this PR is merged by itself, it will leave the `master` branch in a very buggy, unusable state. This PR needs the changes in #55 to function properly. So I'll be closing this and incorporating the changes in this PR to #55.

**Pull request closed without merging**